### PR TITLE
CAN: cleanup drivers and subsystem to remove per instance Kconfig

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

**Review by karstenkoenig — APPROVED** [review]
Always impressed how it keeps working after all these DT macro refactorings! Thanks :-)

**Review by MaureenHelm — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

File: drivers/can/can_mcp2515.c [hunk 1]
```diff
@@ -826,7 +826,7 @@ static int mcp2515_init(struct device *dev)
 	return ret;
 }
 
-#ifdef CONFIG_CAN_1
+#if DT_HAS_DRV_INST(0)
```
Comment (ioannisg) [inline drivers/can/can_mcp2515.c:829]: shouldn't this be inst(1)?

Comment (galak) [reply drivers/can/can_mcp2515.c:829]: No, the first instance is 0.

Comment (ioannisg) [reply drivers/can/can_mcp2515.c:829]: right; is there a reason for the functions right below to have the _1 suffix?

**Review by ioannisg — APPROVED** [review]

File: drivers/can/can_mcp2515.c (hunk 1 above)
Comment (alexanderwachter) [reply drivers/can/can_mcp2515.c:829]: The reason is that the instance is called CAN_1 from the MCU point of view

**Pull request merged**